### DISCUSSION TITLE
How does the automatic to_json - from_json work?

### DISCUSSION THREAD

**Discussion by vassilisw** [nlohmann/json#3262] [Q&A] [answered]
Can anyone explain how [to_json and from_json](https://github.com/nlohmann/json#basic-usage) work in terms of c++?  
I can tell it's not a template specialisation. Is it an implicit conversion of some kind?  
How does the compiler find and use these user defined functions?

Thanks...

**Comment by nlohmann** [comment, marked as answer]
Short answer: ADL (https://en.cppreference.com/w/cpp/language/adl).
